### DISCUSSION TITLE
Creating Contract and Rules for Resource, to offer a Resource

### DISCUSSION THREAD

**Discussion by danniarreza** [International-Data-Spaces-Association/DataspaceConnector#81] [Q&A] [answered]
From the links below, it is stated that an offered resource is only complete if it contains at least one contract offer and at least one representation with at least one artifact:

https://international-data-spaces-association.github.io/DataspaceConnector/Documentation/DataModel

And also stated from the link below that the rule is the object, that contains the usage policy as value:

https://international-data-spaces-association.github.io/DataspaceConnector/CommunicationGuide/Provider

Since we get the endpoint (POST /api/examples/policy) to obtain example policies, which in my case produced this result:
![image](https://user-images.githubusercontent.com/30825334/126064696-e61d059b-c576-41ff-908c-32e67a4b0bd1.png)

Is this result supposed to be attached/pasted to the "value" field of a Rule object that we will relate to a certain contract, or is there any specific value that we should put into the "title" and "value" fields, or how is this usage policy/control should be established?

This question came out, since I attempted to create a Resource offer along with the Contract, Rules, Representation, Artifact and Catalog for that specific resource but I still couldn't manage to get/see the corresponding resource attached to the exposed Catalog through the (POST /api/ids/description) endpoint as shown in the pic below:

![image](https://user-images.githubusercontent.com/30825334/126064842-887a3819-68bc-473c-8581-bb40dc051904.png)

And when I try to request for the Resource offer directly using the Resource id as the elementId, I get this result below 

![image](https://user-images.githubusercontent.com/30825334/126065192-7281f300-3845-4b84-ad4a-7d996b5a415a.png)

However, the console log also gives this hint:

![image](https://user-images.githubusercontent.com/30825334/126065176-bac88a28-f8f2-49f6-b430-ec7e0de455c4.png)

Is there any additional steps required to proceed with testing the IDS way of data sharing?

**Comment by frehwald** [comment]
Hello IDS Team,
I'm experiencing the same issue when converting from connector v4 to v6.0.0 and would appreciate any guidance on the problem with the empty catalog.

**Comment by juliapampus** [comment, marked as answer]
@danniarreza Looks good at first glance. But since I don't know your exact calls and parameters: Have you already taken a look at our scripts and compared if something is missing when adding resources? Did you e.g. probably miss to link the catalog and resource? https://github.com/International-Data-Spaces-Association/DataspaceConnector/tree/main/scripts/tests
